obspy/core/inventory/util.py

```diff
@@ -845,6 +845,15 @@ def _seed_id_keyfunction(x):
     else:
         pass
 
+    # last item is component code, either the original 1-char string, or an int
+    # from 0-2 if any of "ZNE". Python3 does not allow comparison of int and
+    # string anymore (Python 2 always compares ints smaller than any string),
+    # so we need to work around this by making this last item a tuple with
+    # first item False for ints and True for strings.
+    if len(x) == 5:
+        # either a string (has `.__iter__`) or an int (doesn't have it)
+        x[-1] = (hasattr(x[-1], "__iter__"), x[-1])
```

Comment:
Isn't the `int` introduced on purpose on line 842? Why not just create the tuple there in the first place?